kexec.c

```diff
@@ -693,6 +703,34 @@ static void exec_kexec(void)
 	exit(1);
 }
 
+static void do_file_load(char *kernel, char *initrd, char *cmdline)
+{
+	int kernel_fd = -1;
+	int initrd_fd = -1;
+	int ret = 0;
+	unsigned long cmdline_len;
+	unsigned long flags = 0;
+
+	/* cmdline_len includes the '\0'. */
+	cmdline_len = (cmdline == NULL)? 0 : strlen(cmdline) + 1;
+
+	if (kernel)
+		kernel_fd = open(kernel, O_RDONLY);
+
+	if (initrd)
+		initrd_fd = open(initrd, O_RDONLY);
+	else
+		flags |= KEXEC_FILE_NO_INITRAMFS;
+
+	debug_printf("kernel_fd=%d initrd_fd=%d cmdline_len=%lu flags=%lu\n",
+			kernel_fd, initrd_fd, cmdline_len, flags);
+	debug_printf("cmdline=\"%s\"\n", cmdline);
```

Comment:
If cmdline is null, this will blow up.  Maybe you want to have an empty string when cmdline is null?  Depends on what syscall_kexec_file_load expects.